src/Sarif/Writers/SarifLogger.cs

```diff
@@ -62,7 +63,10 @@ public class SarifLogger : IDisposable, IAnalysisLogger
                 }
             }
 
-            var invocation = Invocation.Create(dataToInsert.HasFlag(OptionallyEmittedData.EnvironmentVariables), invocationPropertiesToLog);
+            var invocation = Invocation.Create(
+                emitMachineEnvironment: dataToInsert.HasFlag(OptionallyEmittedData.EnvironmentVariables),
+                emitTimestamps: !dataToRemove.HasFlag(OptionallyEmittedData.NondeterministicProperties),
```

Comment:
>emitTimestamps [](start = 16, length = 14)

Not for now, perhaps, but as you build this out, I wonder whether `Invocation.Create` should just take the `dataToInsert` and `dataToRemove` objects directly, rather than a growing set of individual flags. `Invocation.Create` can do the `HasFlag` checks internally.

Reply:
No, I agree, we should do this. If we're breaking this helper, let's break it once.

---
In reply to: [307786117](https://github.com/microsoft/sarif-sdk/pull/1591#discussion_r307786117) [](ancestors = 307786117)

Reply:
Oops, abort mission. Problem is this help has a mixture of mechanisms to specify what's included, the invocation properties log and these bools. Really should have one converged solution. Let's defer. Agree it needs improvement.

---
In reply to: [307853676](https://github.com/microsoft/sarif-sdk/pull/1591#discussion_r307853676) [](ancestors = 307853676,307786117)